packages/core/src/baseclient.ts

```diff
@@ -653,6 +654,29 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
           throw new SentryError('`beforeSend` returned `null`, will not send event.', 'log');
         }
 
+        const transactionInfo = processedEvent.transaction_info;
+        if (
+          processedEvent.type === 'transaction' &&
+          transactionInfo &&
+          processedEvent.transaction &&
+          processedEvent.transaction !== event.transaction
+        ) {
```

Comment:
Hmmm. Maybe because it's closed, it won't let me use the suggestion feature, but my suggestion would be to add a comment above this, something like `// Check if any of the event processors caused the transaction's name to change, and record that change if so`.

Reply:
Will add, thanks!